### PR TITLE
[Docs] Config Cheat Sheet: SSL_MODE Value Description

### PR DESCRIPTION
**Pull request by bagasme**
This PR add description for all `SSL_MODE` values for both MySQL and PostgreSQL.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 - `SCHEMA`: **\<empty\>**: For PostgreSQL only, schema to use if different from "public". The schema must exist beforehand,
   the user must have creation privileges on it, and the user search path must be set to the look into the schema first 
   (e.g. `ALTER USER user SET SEARCH_PATH = schema_name,"$user",public;`).
-- `SSL_MODE`: **disable**: For PostgreSQL and MySQL only.
+- `SSL_MODE`: **disable**: SSL/TLS encryption mode for connecting to the database. This option is only applied for PostgreSQL and MySQL.
+  - Valid values for MySQL:
+     - `true`: Enable TLS and verify that database server certificate chained up to CA certificate. Ensure that CA certificate chain that signed certificate database server certificate is on system certificate store of both database and Gitea server. See documentation of your system for how to add CA certificate chain to certificate store.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
     - `true`: Enable TLS with verification of the database server certificate against its root certificate. When selecting this option make sure that the root certificate required to validate the database server certificate (e.g. the CA certificate) is on the system certificate store of both the database and Gitea servers. See your system documentation for instructions on how to add a CA certificate to the certificate store.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+     - `false`: Disable TLS.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md:221]:
```suggestion
     - `false`: Disable TLS.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 3]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+     - `disable`: Alias to `false`, for compatibility with PostgreSQL.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
     - `disable`: Alias for `false`, for compatibility with PostgreSQL.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 4]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+     - `skip-verify`: Enable TLS without database server certificate verification. Use this option if you have self-signed or invalid certificate on database server.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
     - `skip-verify`: Enable TLS without database server certificate verification. Use this option if you have self-signed or invalid certificate on the database server.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 5]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+     - `prefer`: Enable TLS with fallback to non-TLS connection.
+  - Valid values for PostgreSQL:
+     - `disable`: Disable TLS.
+     - `require`: Enable TLS without any verifications.
+     - `verify-ca`: Enable TLS and verify that database server certificate chained up to CA certificate.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
     - `verify-ca`: Enable TLS with verification of the database server certificate against its root certificate.
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 6]
```diff
@@ -215,7 +215,18 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+     - `verify-full`: Enable TLS and verify that host name of database server matches either `Subject Alternative Name` or `Common Name` field on database server certificate.
```
Comment (guillep2k) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
     - `verify-full`: Enable TLS and verify the database server name matches the given certificate in either the `Common Name` or `Subject Alternative Name` fields.
```

**Review by guillep2k — COMMENTED** [review]
Some suggestions. I'm not an English native, so I could be a little off too. 😅

**Comment by silverwind** [comment]
Is there a source for these options? Maybe just link to it?

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
Ping LG-TM

**Comment by bagasme** [comment]
@silverwind we use `go-sql-driver/mysql` and `lib/pq` as driver for both database systems. each driver has their own parameters in which we use here.

**Comment by silverwind** [comment]
I would've just linked to their documentation to eliminate the possibility of it going out of date.

**Comment by lafriks** [comment]
It is easier to put it into our docs as we still validate them in our code and to support new values we would anyway need to add support to that ;)

**Pull request merged**